### DISCUSSION TITLE
Update the minimum Postgres version we target (for the metadata database)

### DISCUSSION THREAD

**Discussion by arikfr** [getredash/redash#5933] [General]
We currently target Postgres 9.5 in our builds. This is reflected in our Docker Compose setup, and we have [pinned dependencies](https://github.com/getredash/redash/blob/8487876e7f02d40a7a700557cee35abf1c334902/requirements.txt#L27-L30) to make sure we support Postgres 9.5. This was done because many existing Redash deployments were using this version. 

But Postgres 9.5 was released in 2016 and had its last release in 2021. It is no longer supported (see [Postgres Versioning Policy](https://www.postgresql.org/support/versioning/)). Considering this, it makes no sense to keep targeting this version. We should encourage users to upgrade their Postgres version if they are still using an unsupported version.

The oldest currently supported version of Postgres is 11. It will end its life this year (in November), but we can start with it to make this a gradual change. 

The plan would be:

1. Change Docker Compose setup to use Postgres 11 (in development and tests).
2. Change getredash/setup Docker Compose to use Postgres 15.
3. Update sqlalchemy dependencies.

If anyone got any concerns with this, please share.

**Comment by justinclift** [comment]
As a data point, we've been using PostgreSQL 13 as the Redash metadata database (in production) without issue since November 2021.

So, if PG 15 shows any weirdness for this (unlikely, but still...) then PG 13 is a reasonably safe bet while working out what needs changing. :smile:

**Reply by arikfr** [reply]
I wasn't worried about compatibility issues with newer versions of Postgres. I just didn't want us to rely on features from too-new versions that could cause problems for people using older versions.

**Reply by justinclift** [reply]
No worries. :smile:
